pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -777,6 +777,9 @@ func TestPriorityQueue_Update(t *testing.T) {
 		t.Errorf("Expected: %v after Pop, but got: %v", highPriorityPodInfo.Pod.Name, p.Pod.Name)
 	}
 
+	// call Done for pod returned by Pop
+	q.Done(highPriNominatedPodInfo.Pod.UID)
```

Comment:
It's unclear why we need to call Done right after Pop(). 
I guess we need to do this for L839 - L847. If so, please move all your adding Done()s to L839.

Reply:
Maybe it's time to refactor the entire `TestPriorityQueue_Update`. 😓 It's too complicated and hard to add something further to this test, like this time.
I'd like to see it formed in a table-driven test style.
So, would you mind doing that refactoring in a separate PR after this PR is merged? Or maybe it's OK to include the refactoring in this PR if you prefer to.

Reply:
The preparation and verification of each test case are different. In the test case, I extracted prepareFunc and validateFunc to make the test concise and easy to read.